presto-postgresql/src/main/java/io/prestosql/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -157,12 +255,63 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         if (VARBINARY.equals(type)) {
             return WriteMapping.sliceMapping("bytea", varbinaryWriteFunction());
         }
+        if (TINYINT.equals(type)) {
+            return WriteMapping.longMapping("smallint", tinyintWriteFunction());
```

Comment:
We can't test this with the framework in `TestPostgreSqlTypeMapping`, because Postgres doesn't actually have a `TINYINT` type.  It will get created here as a `SMALLINT` column in Postgres.  Thus when we read it back, we interpret it from the metadata as a `SMALLINT`, but `TestPostgreSqlTypeMapping` will expect a `TINYINT` type back.

Anyways, I will exclude this for now, and do it in another PR sometime.

Reply:
`TestPostgreSqlTypeMapping` has two data setups strategies:
 - `postgresCreateAndInsert` -- table is created in Postgres and populated in postgres
   - write is in Postgres
   - read is in Presto
- `prestoCreateAsSelect` -- table is created and populated in Presto (using CTAS)
   - write is in Presto
   - read is in Presto

The second strategy is the one that you want here, because the method you're fixing (`toWriteMapping`) is called only when Presto is writing.

Reply:
It doesn't matter which you use, there is no `TINYINT` type in Postgres, so it will get created as a `SMALLINT` column in Postgres.  So with this change you can write `TINYINT` via Presto, but when `TestPostgreSqlTypeMapping` reads the data, it will fetch using the column metadata which is of `SMALLINT` type, but the test will expect `TINYINT`.  Thus, I cannot use `DataType.tinyintDataType()` directly.

I added a commit that tests by using a custom `DataType` - you can see whether it's acceptable.